utilities/cli/util.py

```diff
@@ -144,22 +148,90 @@ def fatal(message: str) -> None:
     sys.exit(1)
 
 
+def _get_maybe_sudo() -> str:
+    """Get sudo command if available, with caching to avoid repeated subprocess calls"""
+    global _sudo_available
+
+    if _sudo_available is not None:
+        return _sudo_available
+    try:
+        subprocess.run(["sudo", "--version"], capture_output=True, check=True, timeout=5)
```

Comment:
At a glance it looks like this is adding a one time `sudo --version` check every time a HoloHub CLI command is called that invokes `run_command`.

A quick local check on IGX implies this adds maybe 30-50 ms to command execution, so probably not worth spending effort on. If we do find a need to avoid this extra overhead, could consider checking whether the command needs `sudo` first before polling for `sudo` on the system.